File: Documentation/API/Utilities/CalculateBeatBars.cs
using System;
using RhythmGameUtilities;

const int resolution = 192;

var tempoChanges = new Tempo[]
{
    new() { Position = 0, BPM = 88000 }, new() { Position = 3840, BPM = 112000 },
    new() { Position = 9984, BPM = 89600 }, new() { Position = 22272, BPM = 112000 },
    new() { Position = 33792, BPM = 111500 }, new() { Position = 34560, BPM = 112000 },
    new() { Position = 42240, BPM = 111980 }
};

var beatBars = Utilities.CalculateBeatBars(tempoChanges, resolution, true);

Console.WriteLine(beatBars.Length); // 440

File: Documentation/API/Utilities/CalculateBeatBars.js
import RhythmGameUtilities from '@rhythm-game-utilities/core';

const tempoChanges = [
  { position: 0, bpm: 88000 },
  { position: 3840, bpm: 112000 },
  { position: 9984, bpm: 89600 },
  { position: 22272, bpm: 112000 },
  { position: 33792, bpm: 111500 },
  { position: 34560, bpm: 112000 },
  { position: 42240, bpm: 111980 }
];

const resolution = 192;

const beatBars = RhythmGameUtilities.CalculateBeatBars(tempoChanges, resolution, true);

console.log(beatBars.length); // 440
